ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/MaximumLagFilter.java

```diff
@@ -66,8 +66,8 @@ public boolean filter(final KsqlHostInfo hostInfo) {
           final long offsetLag = Math.max(endOffset - hostLag.getCurrentOffsetPosition(), 0);
           return offsetLag <= allowedOffsetLag;
         })
-        // If we don't have lag info, we'll be conservative and not include the host
-        .orElse(false);
+        // If we don't have lag info, we'll be conservative and include the host
```

Comment:
I think we should do this and then prime the cache on the remote host on-demand as we discussed..  This will ensure correct filtering in all cases.. 

Choosing to ignore nodes if we don't have lag , would mean that, on every bounce of the ksql server, queries will fail for few seconds until lags are propogated across the cluster again (bounced server will begin with an empty map).. Thus a rolling bounce of the cluster will see large error spikes.. This is a bad move IMO

Reply:
Please file an issue for the the follow up work of priming the cache on the remote host, on demand...

Reply:
Yeah, I agree with that.  I guess the only downside is that if we don't know the lag and we forward the request to that host, we may not be honoring the max lag configuration and may be returning a really stale value.

Reply:
What I meant was, the remote host (i.e the one whose lag we dont know) can do the lag enforcement and fail the query and send an error response.. This lag exchange control plane is an optimization for not doing this round trip when we know the host is backed up/lagging significantly

Reply:
#4471 #4472 - Those are the two followup issues.

I see.  That makes sense.  I agree that it seems necessary to ensure the filtering if the client, in some way, is relying on that for some notion of consistency, etc.  If we're going to enforce this on the lagging host and the control plane is an optimization, then this makes sense.